lib/utils/color.dart

```diff
@@ -8,14 +8,9 @@ import 'package:flutter/material.dart';
 /// Inspired by https://stackoverflow.com/a/13532993
 Color shadeColor(double factor, Color color, {int minBit = 0, int maxBit = 0xff}) {
   assert(factor >= -1 && factor <= 1);
-  var r = ((color.value >> 16) & 0xff).clamp(0, 0xff);
-  var g = ((color.value >> 8) & 0xff).clamp(0, 0xff);
-  var b = (color.value & 0xff).clamp(0, 0xff);
-
   factor = (1 + factor);
-  r = (r * factor).toInt().clamp(minBit, maxBit);
-  g = (g * factor).toInt().clamp(minBit, maxBit);
-  b = (b * factor).toInt().clamp(minBit, maxBit);
-
-  return Color((0xff << 24) + (r << 16) + (g << 8) + b);
+  final r = (color.r * 255.0 * factor).toInt().clamp(minBit, maxBit);
+  final g = (color.g * 255.0 * factor).toInt().clamp(minBit, maxBit);
+  final b = (color.b * 255.0 * factor).toInt().clamp(minBit, maxBit);
+  return Color.fromARGB(0xFF, r, g, b);
```

Comment:
Can you make an issue to check whether we should put an original alpha color here instead? I'm not sure I did this intentionally or not when I implemented this

Reply:
In my opinion it should not change the opacity. Let's use the opacity form he color argument, and let's update the golden test images and see if we have any changes.

Reply:
Yes, but I would prefer if we did that in a separate issue